### PR TITLE
use babel-minify for production builds to avoid incorrect mangling of JS

### LINKED ISSUE #4842: Switching to PR branches (forks) no longer works
## Description
Selecting a pull request under the "Current Branch" menu _used_ to stash changes and check out the PR locally. In the case of a PR that was opened against a fork (rather than a direct branch), GitHub Desktop used to create a new remote usually prefixed with `github-desktop-xxxxx`.

In the two most recent (stable) releases, the functionality to switch to PR branches silently fails with no change in the UI nor any visible errors in the logs.

Shown below is my selection of a PR hosted on GitHub (in a private repository) where the selection (via mouse click) closes the menu but results in no branch switch:

record

## Version
* GitHub Desktop: 1.2.x (occurs in 1.2.2 and 1.2.1)
* Operating system: MacOS 10.13.4

## Steps to Reproduce

1. Add a local repository
2. Create a pull request where the PR originates in a fork (not a branch of the repository)
3. Use GitHub Desktop to view and checkout this PR

### Expected Behavior
PRs originating from forks are available for local checkout (pull)

### Actual Behavior
PRs originating from forks are _not_ available for local checkout (pull)

### Logs

[2018-05-31.desktop.production.log](https://github.com/desktop/desktop/files/2059746/2018-05-31.desktop.production.log)

### PR DESCRIPTION
**Pull request by shiftkey**
Resolving the underlying issue(s) with `uglify-es` that we unfortunately stumbled upon.

 - [x] confirm problem JS from #4848 is not emitted
 - [x] confirm problem JS from #4859 is not emitted
 - [x] install test build and confirm features from #4848 and #4859 (i.e. can select a co-author and can checkout a fork PR) are working

Fixes #4842

Feel free to use [`reproduce-minification-issue`](https://github.com/desktop/desktop/compare/reproduce-minification-issue) branch (this is based off `release-1.2.2`) to experiment with reproducing the isolated version of the problem against `uglify-es` to help with reporting this upstream. Annotated commits are mentioned in https://github.com/desktop/desktop/pull/4859#issuecomment-394564362.

### CODE REVIEW

**Comment by shiftkey** [comment]
Tested this out on Windows - if someone wants to test the Mac version as part of reviewing this that'd be 💖

**Comment by iAmWillShepherd** [comment]
@shiftkey, testing this on macOS now.

File: app/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "productName": "GitHub Desktop",
   "bundleID": "com.github.GitHubClient",
   "companyName": "GitHub, Inc.",
-  "version": "1.2.2",
+  "version": "1.2.3-test1",
```
Comment (iAmWillShepherd) [inline app/package.json]: Will this version number need to be updated?

Comment (shiftkey) [reply app/package.json]: Yes, this should be updated as part of generating the changelog for the next version.

**Pull request merged**